refactor/actions.py

```diff
@@ -195,11 +195,59 @@ def _stack_effect(self) -> tuple[ast.AST, int]:
         return (self.node, 1)
 
 
+@_hint("deprecated_alias", "NewStatementAction")
+@dataclass
+class LazyInsertBefore(_LazyActionMixin[ast.stmt, ast.stmt]):
+    """Inserts the re-synthesized version :py:meth:`LazyInsertBefore.build`'s
+    output right before the given `node`.
+
+    .. note::
+        Subclasses of :py:class:`LazyInsertBefore` must override
+        :py:meth:`LazyInsertBefore.build`.
+
+    .. note::
+        This action requires both the `node` and the built target to be statements.
+    """
+
+    def apply(self, context: Context, source: str) -> str:
+        lines = split_lines(source, encoding=context.file_info.get_encoding())
+        indentation, start_prefix = find_indent(
+            lines[self.node.lineno - 1][: self.node.col_offset]
+        )
+
+        replacement = split_lines(context.unparse(self.build()))
+        replacement.apply_indentation(indentation, start_prefix=start_prefix)
+
+        original_node_start = cast(int, self.node.lineno)
+        if lines[original_node_start].endswith(lines._newline_type):
+            replacement[-1] += lines._newline_type
+        else:
+            # If the original anchor's last line doesn't end with a newline,
+            # then we need to also prevent our new source from ending with
+            # a newline.
+            replacement[0] = lines._newline_type + replacement[0]
```

Comment:
Do we need this check? This was solely done for preserving the end on the InsertAfter but it seems redundant for InsertBefore. In either case we should only add the new statement right before the original one without any sort of line handling.